src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,257 @@
+
+        if (answers.length > 0){
+            let resultIds = [...new Set(answers.map(a => a.surveyResultID ))];
+            resultIds.forEach( (resultId) => {
+                var resultsForSubmission = answers.filter(a => a.surveyResultID == resultId);
+                var bpLogDateResults = resultsForSubmission.find( r => r.resultIdentifier == props.dateOfResultId);
+                var bpLogDate =bpLogDateResults?.answers[0];
+                var bpSystolicResults = resultsForSubmission.find( r => r.resultIdentifier == props.systolicResultId);
+                var bpSystolic = bpSystolicResults?.answers[0];
+                var bpDiastolicResults = resultsForSubmission.find( r => r.resultIdentifier == props.diastolicResultId);
+                var bpDiastolic = bpDiastolicResults?.answers[0];
+
+                if (bpLogDate){
+                    var useDate = startOfDay(new Date(bpLogDate));
+                    var formattedDate = format(useDate, "MM/dd");
+                    var newBpEntry : IBloodPressureDataPoint = {
+                        dateLabel: formattedDate, 
+                        date: useDate,
+                        systolic : Number(bpSystolic ?? 0), 
+                        diastolic: Number(bpDiastolic ?? 0)
+                    };
+
+                    bpDataPoints.push(newBpEntry);
+                }
+            });
+        }
+
+        return bpDataPoints;
+    }
+
+    function createDumbbellsPerDay(bpDataPoints : IBloodPressureDataPoint[]){
+        var dbs : IBloodPressureDumbbell[] = [];
+        if (bpDataPoints.length > 0){
+            let logDates = [...new Set(bpDataPoints.map(dp => dp.dateLabel))];
+            logDates.forEach((logDay) =>{
+                const entriesForDate = bpDataPoints.filter( a => a.dateLabel == logDay);
+                const systolicEntriesForDate = entriesForDate.map( e => e.systolic);
+                const systolicInterval = buildInterval(systolicEntriesForDate);
+                const systolicAverage = systolicEntriesForDate.reduce((a, b) => a + b) / systolicEntriesForDate.length;
+                const diastolicEntriesForDate = entriesForDate.map( e => e.diastolic);
+                const diastolicInterval = buildInterval(diastolicEntriesForDate);
+                const diastolicAverage = diastolicEntriesForDate.reduce((a, b) => a + b) / diastolicEntriesForDate.length;
+                const dataPoint : DataPoint = {dataSet1: diastolicInterval, dataSet2: systolicInterval};
+                var db : Dumbbell = {dataPoint  :dataPoint, xValue : logDay, class : assignClass(systolicAverage, diastolicAverage)};
+                dbs.push({date: entriesForDate[0].date, dumbbell: db, averageSystolic : systolicAverage, averageDiastolic : diastolicAverage});
+            });
+        }
+
+        return dbs;
+    }
+
+    async function getSurveyAnswers(surveyName: string, stepId : string[], resultId : string[]): Promise<SurveyAnswer[]> {
```

Comment:
This is in the context of a component that has props that declare the step / result identifiers...you do not need to pass them around as parameters.  Just use the props